librz/asm/arch/hexagon/hexagon.h

```diff
@@ -93,33 +75,65 @@ typedef struct {
+	HexInsnID identifier; ///< The instruction identifier
+	char text_infix[128]; ///< Textual disassembly of the instruction.
+	HexOp ops[HEX_MAX_OPERANDS]; ///< The operands of the instructions.
 } HexInsn;
 
+/**
+ * \brief The instruction container holds one instruction or two sub-instructions if it is a duplex.
+ * It stores meta information about those instruction(s) like opcode, packet information or the parse bits.
+ */
+typedef struct {
+	ut8 parse_bits; ///< Parse bits of instruction.
+	bool is_duplex; ///< DOes this container hold two sub-instructions?
+	ut32 identifier; ///< Equals instruction ID if is_duplex = false. Otherwise: (high.id << 16) | (low.id & 0xffff)
+	union {
+		HexInsn *sub[2]; ///< Pointer to sub-instructions if is_duplex = true. sub[0] = high, sub[1] = low
+		HexInsn *insn; ///< Pointer to instruction if is_duplex = false.
+	} bin;
+	ut32 addr; ///< Address of container. Equals address of instruction or of the high sub-instruction if this is a duplex.
+	ut32 opcode; ///< The instruction opcode.
+	HexPktInfo pkt_info; ///< Packet related information. First/last instr., prefix and postfix for text etc.
+	RzAsmOp asm_op; ///< Private copy of AsmOp. Currently only of interest because it holds the utf8 flag.
+	RzAnalysisOp ana_op; ///< Private copy of AnalysisOp. Analysis info is written into it.
+	char text[296]; ///< Textual disassembly
+} HexInsnContainer;
+
+/**
+ * \brief Represents an Hexagon instruction packet.
+ * We do not assign instructions to slots, but the order of instructions matters nonetheless.
+ * The layout of a packet is:
+ *
+ * low addr | Slot 3
+ * ---------+----------
+ *          | Slot 2
+ * ---------+----------
+ *          | Slot 1    -> High Sub-Instruction is always in Slot 1
+ * ---------+----------
+ * high addr| Slot 0    -> Low Sub-Instruction is always in Slot 0
+ *
+ * Because of this order the textual disassembly of duplex instructions is: "<high-text> ; <low-text>".
+ * Also, the high sub-instruction is located at the _lower_ memory address (aligned to 4 bytes).
+ * The low sub-instruction at <high.addr + 2>.
+ *
+ * This said: The HexPkt.bin holds only instruction container, no instructions!
+ * The container holds the instructions or sub-instructions.
+ */
 typedef struct {
-	RzList *insn; // List of instructions.
-	bool last_instr_present; // Has an instruction the parsing bits 0b11 set (is last instruction).
-	bool is_valid; // Is it a valid packet? Do we know which instruction is the first?
-	ut32 hw_loop0_addr; // Start address of hardware loop 0
-	ut32 hw_loop1_addr; // Start address of hardware loop 1
-	ut64 last_access; // Last time accessed in milliseconds
-	ut32 pkt_addr; // Address of the packet. Equals the address of the first instruction.
-	bool is_eob; // Is this packet the end of a code block? E.g. contains unconditional jmp.
+	RzList /* HexInsnContainer */ *bin; ///< Descending by address sorted list of instruction containers.
```

Comment:
Here I recommend using RzVector instead, not for this PR, though.

Reply:
https://github.com/rizinorg/rz-hexagon/issues/76